File: cpp/acronym/acronym.cpp
#include "acronym.h"

#include <cctype>

namespace acronym {

std::string acronym(const std::string &str) {
    std::string ret;
    size_t pos = 0;

    while (pos < str.size()) {
        while (pos < str.size() && !std::isalpha(str[pos])) {
            ++pos;
        }

        ret.push_back(std::toupper(str[pos]));

        while (pos < str.size() && std::isalpha(str[pos])) {
            ++pos;
        }
    }

    return ret;
}

} // namespace acronym

File: go/acronym/acronym.go
// This is a "stub" file.  It's a little start on your solution.
// It's not a complete solution though; you have to write some code.

// Package acronym should have a package comment that summarizes what it's about.
// https://golang.org/doc/effective_go.html#commentary
package acronym

import (
	"regexp"
	"unicode"
)

var re = regexp.MustCompile(`[a-zA-Z][a-zA-Z']*`)

// Abbreviate should have a comment documenting it.
func Abbreviate(s string) string {
	matches := re.FindAllString(s, -1)

	ret := ""
	for _, match := range matches {
		r := unicode.ToUpper(rune(match[0]))
		ret += string(r)
	}

	return ret
}
